src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -66,6 +66,23 @@ public override Conversion GetMethodGroupDelegateConversion(BoundMethodGroup sou
                 useSiteInfo.Add(memberUseSiteInfo);
             }
 
+            // If synthesizing a delegate with `decimal` default value, check that `DecimalConstantAttribute` is available.
+            if (methodSymbol.OriginalDefinition is SynthesizedDelegateInvokeMethod invoke)
+            {
+                foreach (var p in invoke.Parameters)
+                {
+                    var defaultValue = p.ExplicitDefaultConstantValue;
+                    if (defaultValue != ConstantValue.NotAvailable &&
+                        defaultValue.SpecialType == SpecialType.System_Decimal)
+                    {
+                        Binder.GetWellKnownTypeMember(Compilation,
```

Comment:
>GetWellKnownTypeMember [](http://example.com/codeflow?start=31&length=22)

Consider explicitly passing result of ```WellKnownMembers.IsSynthesizedAttributeOptional(...)```  to this method. As ReportUseSiteDiagnosticForSynthesizedAttribute does.